File: evals/fixtures/pagination-window-fix/src/window.js
"use strict";

/**
 * pageBounds(total, perPage, page) -> the `[startIndex, endIndex)` half-open
 * window for a 1-based `page` over `total` items.
 *
 * Spec: both indices must be clamped to `total`, so the final (partial) page
 * reports an in-range window and a page past the end reports an empty window
 * `[total, total]` — never one that overruns the data.
 *
 * BUG: `end` (and, past the end, `start`) is computed without clamping to
 * `total`, so the last page overruns. Clamp both indices to `total`.
 */
function pageBounds(total, perPage, page) {
  const start = (page - 1) * perPage;
  const end = page * perPage;
  return [start, end];
}

module.exports = { pageBounds };

File: evals/fixtures/go-pagination-window/window.go
package paginationwindow

// Window returns the half-open page [offset, offset+limit), clamped to items.
func Window[T any](items []T, offset int, limit int) []T {
	if offset < 0 {
		offset = 0
	}
	if limit <= 0 || offset >= len(items) {
		return []T{}
	}
	end := offset + limit
	// BUG: exact-end pages are shortened by one item.
	if end >= len(items) {
		end = len(items) - 1
	}
	return items[offset:end]
}
